### PR TITLE
Issue #2352 - WIP: historyarchive: Retry on http 503 response

### PR DESCRIPTION
**Pull request by nyetwurk**
POC fix for #2352

WIP: Retry forever on 503
1) should give up eventually
2) should do a backoff

Please don't merge, this is just an example of how this could be fixed.

### CODE REVIEW

File: support/historyarchive/http_archive.go [hunk 1]
```diff
@@ -38,12 +38,20 @@ func (b *HttpArchiveBackend) GetFile(pth string) (io.ReadCloser, error) {
...
+				resp.Body.Close()
+			}
+			return nil, err
 		}
-		return nil, err
+		if resp.StatusCode != 503 {
+			break
+		}
+		// FIXME: exponential backoff
+		// sleep(exponential_backoff)
```
Comment (nyetwurk) [inline support/historyarchive/http_archive.go]: 1) should sleep with exponential pseudo-random backoff
2) should eventually give up

Comment (nyetwurk) [reply support/historyarchive/http_archive.go]: Also, this might be better done in `.Do()` itself, since there are other things that can cause a 503

**Comment by 2opremio** [comment]
Do you still want to pursue this or shall we close it?

**Comment by nyetwurk** [comment]
I was hoping someone would pick up where I left off, it still affects us.

**Comment by leighmcculloch** [comment]
I think the problem can be solved but we probably should take another route. @bartekn and I commented over on the issue with some thoughts. Let's discuss there and see what we can find. See https://github.com/stellar/go/issues/2352#issuecomment-700960427.

**Pull request closed without merging**